### ISSUE TITLE
ciscoise_node_services_profiler_probe_config - Resource creation not working

### ISSUE THREAD

**Issue by ragu2k8** [CiscoISE/terraform-provider-ciscoise#15]
@wastorga  when we run below resource, its not updated on server side(probes are not enabled) and its not showing any error on terraform either
however direct api works without issues.

```
resource "ciscoise_node_services_profiler_probe_config" "configure_profiling_probe" {
  parameters {
    hostname           = "isenode1"
    active_directory   {
      days_before_rescan = 1
    }
    http  {
      interfaces  {
        interface  = "GigabitEthernet 0"
      }
    }
    dhcp  {
      interfaces  {
        interface  = "GigabitEthernet 0"
      }
      port = 67
    }
    radius = []
    nmap = []
    dns {
      timeout =  2
    }
    snmp_query {
      retries = 2
      timeout = 1000
      event_timeout = 30
    }
  }
}
```

log:
Changes to Outputs:
```
  + ciscoise_node_services_profiler_probe_config_details = {
      + id           = (known after apply)
      + item         = (known after apply)
      + last_updated = (known after apply)
      + parameters   = [
          + {
              + active_directory = [
                  + {
                      + days_before_rescan = 1
                    },
                ]
              + dhcp             = [
                  + {
                      + interfaces = [
                          + {
                              + interface = "GigabitEthernet 0"
                            },
                        ]
                      + port       = 67
                    },
                ]
              + dhcp_span        = []
              + dns              = [
                  + {
                      + timeout = 2
                    },
                ]
              + hostname         = "isenode1"
              + http             = [
                  + {
                      + interfaces = [
                          + {
                              + interface = "GigabitEthernet 0"
                            },
                        ]
                    },
                ]
              + netflow          = []
              + nmap             = []
              + pxgrid           = null
              + radius           = []
              + snmp_query       = [
                  + {
                      + event_timeout = 30
                      + retries       = 2
                      + timeout       = 1000
                    },
                ]
              + snmp_trap        = []
            },
        ]
    }
ciscoise_node_services_profiler_probe_config.configure_profiling_probe: Creating...
ciscoise_node_services_profiler_probe_config.configure_profiling_probe: Creation complete after 0s [id=hostname:=isenode1]

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.

```

**Comment by wastorga** [comment]
It does not update it the first time. The first time only adds it in Terraform as a resource.
Following changes to the resource.tf settings will call the update ISE.

**Comment by ragu2k8** [comment]
@wastorga thanks for clarifying. however when i do changes for below probes, its not getting updated.
radius = []
nmap = [] 
above just need empty list right nothing requried to pass
its not being updated

**Comment by wastorga** [comment]
@ragu2k8 
I lack the means of testing for the moment. However, I suspect it is related to the `isEmptyValue` function, which does not allow the empty array/slice to be added in the update context. I will look into it as soon as I can.

**Comment by ragu2k8** [comment]
sure @wastorga let me know. I did bit more testing and i can add more details if you need.

**Comment by ragu2k8** [comment]
@wastorga  any luck?

**Comment by wastorga** [comment]
Some, @ragu2k8.
The API's empty probes (i.e., without properties to set) behave more like boolean (true if it has an array, false if null).
So, considering Terraform Default behavior, I am tempted to change them to a TypeString with enum values ["", true, false], where the empty string represents that it does not want any change.
I'll get back to you tomorrow with an update.

**Comment by ragu2k8** [comment]
sure @wastorga thanks for update

**Comment by wastorga** [comment]
@ragu2k8 
I released the new version, v0.1.0-rc.3, a couple of minutes ago. The resource (node_services_profiler_probe_config) now performs the update at Tf (resource) creation. As discussed earlier, it also has new parameters' types for Nmap, pxGrid, and Radius.

**Comment by ragu2k8** [comment]
@wastorga  awesome..i have just tested it and it works fine. Thanks for support.
